scene/3d/gpu_particles_3d.cpp

```diff
@@ -92,7 +92,11 @@ void GPUParticles3D::set_one_shot(bool p_one_shot) {
 
 	if (is_emitting()) {
 		if (!one_shot) {
-			restart();
+			if (!use_fixed_seed) {
+				set_seed(Math::rand());
```

Comment:
I just realized, we may not want to use `set_seed` here as it will change the value that gets saved if `seed` is serialized. 

We probably just want to call the renderingserver function directly

Reply:
Hey @clayjohn i see you approved and left a review. Maybe it's just a bug on my end but it says you still approve. Do you want me to make these changes you suggested?

Reply:
Got back to my PC this morning.  I'll go update this as well :)

Reply:
Examining the code, I'm actually not sure about this change.  I see there is a public ``get_seed`` function.  So if I were to call ``RS::get_singleton()->particles_set_seed`` without also calling ``seed = p_seed;``, this would make ``gpu_particles_3d.cpp`` out of sync with the rendering server.  The ``get_seed`` function would no longer correctly reflect the value that has been set on the rendering server.  What are your thoughts on this?  @QbieShay Do you have some input as well?

Reply:
We'll send it as-is; it'll be easy to apply a fix if this does cause a desync.